src/imex/export/useExportTable.ts

```diff
@@ -21,39 +27,39 @@ export const useExportTable = <D extends { [key: string]: any } = any>(
   paramsRef.current = params
 
   const downloadTableData = React.useCallback(
-    (title: string, options: Partial<UseExportTableParams<D>> = {}) => {
-      const { data = [], columns, type = 'csv', ...exportOptions } = {
+    (title: string, options?: Partial<UseExportTableParams<D>>) => {
+      const { data, columns, ...exportOptions } = {
+        data: [],
+        type: 'xls',
         ...paramsRef.current,
         ...options,
-      }
+      } as const
 
       const imexColumns = getImexColumns(columns)
       const imexData = data.map((row) =>
-        imexColumns.map(({ accessor, meta }) => {
-          let value = get(row, accessor as string)
-          const parse = (v: any) => {
-            if (
-              !meta?.imex?.parse &&
-              ['string[]', 'number[]'].includes(
-                meta?.imex?.type as 'string[]' | 'number[]'
-              )
-            ) {
-              return v?.join(',')
-            }
-            return meta?.imex?.parse?.(v, Object.values(row)) ?? v
-          }
-          if (type === 'xls' && meta?.imex?.type === 'number') {
-            return isNumber(value) ? Number(parse(value)) : value
-          }
-          return parse(value)
+        imexColumns.map((column) => {
+          const meta = column.meta?.imex
+          const rawValue = get(row, column.accessor as string)
+          const valueType = meta?.type
+
+          const value = meta?.parse
+            ? meta.parse(rawValue, Object.values(row))
+            : ARRAY_TYPES.has(valueType!) && Array.isArray(rawValue)
```

Comment:
😬  You find nested ternary clearer?

Reply:
As long as the condition are clear, yes I do :man_shrugging:

What do you want exactly:
```js
let value = get(row, column.accessor as string)

if (meta?.parse) {
  value = meta.parse(value, Object.values(row))
} else if (ARRAY_TYPES.has(valueType!) && Array.isArray(rawValue)) {
  value = value.join(',')
}
```